skimage/measure/_marching_cubes_lewiner_cy.pyx

```diff
@@ -0,0 +1,1463 @@
+        self.TILING7_4_2 = Lut(TILING7_4_2)
+        self.TILING8 = Lut(TILING8)
+        self.TILING9 = Lut(TILING9)
+        self.TILING10_1_1 = Lut(TILING10_1_1)
+        self.TILING10_1_1_ = Lut(TILING10_1_1_)
+        self.TILING10_1_2 = Lut(TILING10_1_2)
+        self.TILING10_2 = Lut(TILING10_2)
+        self.TILING10_2_ = Lut(TILING10_2_)
+        self.TILING11 = Lut(TILING11)
+        self.TILING12_1_1 = Lut(TILING12_1_1)
+        self.TILING12_1_1_ = Lut(TILING12_1_1_)
+        self.TILING12_1_2 = Lut(TILING12_1_2)
+        self.TILING12_2 = Lut(TILING12_2)
+        self.TILING12_2_ = Lut(TILING12_2_)
+        self.TILING13_1 = Lut(TILING13_1)
+        self.TILING13_1_ = Lut(TILING13_1_)
+        self.TILING13_2 = Lut(TILING13_2)
+        self.TILING13_2_ = Lut(TILING13_2_)
+        self.TILING13_3 = Lut(TILING13_3)
+        self.TILING13_3_ = Lut(TILING13_3_)
+        self.TILING13_4 = Lut(TILING13_4)
+        self.TILING13_5_1 = Lut(TILING13_5_1)
+        self.TILING13_5_2 = Lut(TILING13_5_2)
+        self.TILING14 = Lut(TILING14)
+        
+        self.TEST3 = Lut(TEST3)
+        self.TEST4 = Lut(TEST4)
+        self.TEST6 = Lut(TEST6)
+        self.TEST7 = Lut(TEST7)
+        self.TEST10 = Lut(TEST10)
+        self.TEST12 = Lut(TEST12)
+        self.TEST13 = Lut(TEST13)
+        
+        self.SUBCONFIG13 = Lut(SUBCONFIG13)
+
+
+def marching_cubes(im, double isovalue, LutProvider luts, int st=1, int classic=0):
+    """ marching_cubes(im, double isovalue, LutProvider luts, int st=1, int classic=0)
+    This is the main entry to apply marching cubes.
+    Returns (vertices, faces, normals, values)
+    """ 
+    
+    # Typdedef image
+    cdef np.ndarray[FLOAT32_T, ndim=3] im_ = im
+    
+    # Get dimemsnions
+    cdef int Nx, Ny, Nz
+    Nx, Ny, Nz = im.shape[2], im.shape[1], im.shape[0]
+    
+    # Create cell to use throughout
+    cdef Cell cell = Cell(luts, Nx, Ny, Nz)
+    
+    # Typedef variables
+    cdef int x, y, z
+    cdef int nt
+    cdef int case, config, subconfig
+    
+    # Unfortunately, using a step in the for loops degregades performance.
+    # Quick and dirty trick below ...
+    
+    if st == 1:
+        
+        for z in range(0,Nz-1):
+            cell.new_z_value() # Indicate that we enter a new layer
+            for y in range(0,Ny-1):
+                for x in range(0,Nx-1):
+                    
+                    # Initialize cell
+                    cell.set_cube(isovalue, x, y, z, st,
+                        im_[z   ,y, x], im_[z   ,y, x+st], im_[z   ,y+st, x+st], im_[z   ,y+st, x],
+                        im_[z+st,y, x], im_[z+st,y, x+st], im_[z+st,y+st, x+st], im_[z+st,y+st, x] )
+                    
+                    # Do classic!
+                    if classic:
+                        # Determine number of vertices
+                        nt = 0
+                        while luts.CASESCLASSIC.get2(cell.index, 3*nt) != -1:
+                            nt += 1
```

Comment:
Would this be equivalent? If so, it would save you a lot of multiplication in the innermost loop.
``` python
nt = 0
while luts.CASESCLASSIC.get2(cell.index, nt) != -1:
    nt += 3
```